src/freenet/crypt/HMAC.java

```diff
@@ -53,6 +54,10 @@ public static boolean verify(HMAC hash, byte[] key, byte[] data, byte[] mac) {
     return MessageDigest.isEqual(mac, mac(hash, key, data));
   }
 
+  public static boolean verify(HMAC hash, byte[] key, byte[] data, byte[] mac, int length) {
```

Comment:
Or better yet, keep the truncated key in the sessionKey object and don't change the API